src/main/scala/exercises/04_Options.scala

```diff
@@ -34,7 +34,12 @@ object Options {
    * - does not exist: 					"not existing"
    */
   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
+    rooms.get(room) match {
+      case Some(Some("locked")) => "not available"
+      case Some(Some(n)) => n
+      case Some(None) => "empty"
+      case None => "not existing"
+    }
```

Comment:
You can also use `rooms.getOrElse(key, default)` to get the status of the room as an `Option[String]`, rather than an `Option[Option[String]]`. I'm not sure that's necessarily better, but it does save you having to write `Some(Some(n))`, which I found a little confusing.